specification/managementgroups/resource-manager/Microsoft.Management/preview/2024-02-01-preview/management.json

```diff
@@ -0,0 +1,2383 @@
+          "type": "string",
+          "readOnly": true,
+          "description": "The fully qualified ID for the child resource (management group or subscription).  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the child entity."
+        },
+        "displayName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The friendly name of the child resource."
+        },
+        "children": {
+          "description": "The list of children.",
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CreateManagementGroupChildInfo"
+          }
+        }
+      }
+    },
+    "CreateParentGroupInfo": {
+      "description": "(Optional) The ID of the parent management group used during creation.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified ID for the parent management group.  For example, /providers/Microsoft.Management/managementGroups/0000000-0000-0000-0000-000000000000"
+        },
+        "name": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The name of the parent management group"
+        },
+        "displayName": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The friendly name of the parent management group."
+        }
+      }
+    },
+    "Permissions": {
+      "type": "string",
+      "enum": [
+        "noaccess",
+        "view",
+        "edit",
+        "delete"
+      ],
+      "x-ms-enum": {
+        "name": "Permissions",
+        "modelAsString": false,
+        "values": [
+          {
+            "value": "noaccess"
+          },
+          {
+            "value": "view"
+          },
+          {
+            "value": "edit"
+          },
+          {
+            "value": "delete"
+          }
+        ]
+      },
+      "description": "The users specific permissions to this item."
+    },
+    "CheckNameAvailabilityRequest": {
+      "description": "Management group name availability check parameters.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "the name to check for availability"
+        },
+        "type": {
+          "type": "string",
+          "enum": [
+            "Microsoft.Management/managementGroups"
+          ],
+          "x-ms-enum": {
+            "name": "Type",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "Microsoft.Management/managementGroups"
+              }
+            ]
+          },
+          "description": "fully qualified resource type which includes provider namespace"
+        }
+      }
+    },
+    "PutApplication": {
```

Comment:
>PutApplication [](http://example.com/codeflow?start=5&length=14)

It's recommended (but not required) to use the same schema for PUT request and response and GET response, even though some properties aren't used on create. Generally, GET-modify-PUT should be a supported pattern. You should be able to eliminate this definition and just use Application for the PUT request